### Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. 

Какова вероятность того, что в первый месяц выйдут из строя: 

а). все детали 

б). только две детали 

в). хотя бы одна деталь 

г). от одной до двух деталей?



In [1]:
from math import factorial

p1 = 0.1
q1 = 0.9
p2 = 0.2
q2 = 0.8
p3 = 0.25
q3 = 0.75

A = p1 * p2 * p3 # Вероятность, что все детали выйдут из строя (задача А)
X = q1 * q2 * q3 # Ни одна деталь не выйдет из строя
B = (p1 * p2 * q3) + (p1 * q2 * p3) + (q1 * p2 * p3) # Только две детали выйдут из строя (задача Б)
C = 1 - X # Хотя бы одна деталь выйдет из строя (задача В)
Y = (p1 * q2 * q3) + (q1 * p2 * q3) + (q1 * q2 * p3) # Только одна деталь выйдет из строя
D = Y + B # От одной до двух деталей выйдет из строя (задача Г)

A, B, C, D

(0.005000000000000001, 0.08000000000000002, 0.45999999999999996, 0.455)

Вероятность того, что в первый месяц:

* выйдут из строя все детали = 0.5%
* выйдут из строя только две детали = 8%
* выйдет из строя хотя бы одна деталь = 46%
* выйдет из строя от одной до двух деталей = 45.5%